### PR TITLE
MM-9866 Fixing unit tests for Go 1.10

### PR DESCRIPTION
**Pull request by crspeller**
Enterprise changes: https://github.com/mattermost/enterprise/pull/280

### CODE REVIEW

File: api4/file_test.go [hunk 1]
```diff
@@ -359,7 +359,8 @@ func TestGetFileHeaders(t *testing.T) {
 	t.Run("js", testHeaders(data, "test.js", "text/plain", false))
 	t.Run("go", testHeaders(data, "test.go", "application/octet-stream", false))
 	t.Run("zip", testHeaders(data, "test.zip", "application/zip", false))
-	t.Run("exe", testHeaders(data, "test.exe", "application/x-ms", false))
+	// Not every platform can recognize these
+	//t.Run("exe", testHeaders(data, "test.exe", "application/x-ms", false))
```
Comment (jespino) [inline api4/file_test.go]: Here we can add the ".exe" extension to the mime types with mime.AddExtensionType. Probably in the `model/file_info.go` file.

File: Makefile [hunk 2]
```diff
@@ -360,6 +361,7 @@ ifeq ($(BUILD_ENTERPRISE_READY),true)
 endif
 
 test-server: test-te test-ee ## Runs tests.
+	find . -type d -name data -not -path './vendor/*' | xargs rm -rf
```
Comment (jespino) [inline Makefile]: I'm not sure about it. Running the unit tests remove any data directory in mattermost-server repo? I'm not sure if it's something safe.

Comment (crspeller) [reply Makefile]: After running the tests it creates a bunch of data directories from the tests. This just deletes them.

File: api4/file_test.go (hunk 1 above)
Comment (crspeller) [reply api4/file_test.go]: I guess, but I don't think it's important that we recognize those properly. In fact, it's probably better if we don't.

**Pull request merged**